backend/app/rest/residents_routes.py

```diff
@@ -0,0 +1,66 @@
+from flask import Blueprint, current_app, jsonify, request
+from ..middlewares.auth import require_authorization_by_role
+from ..services.implementations.residents_service import ResidentsService
+import json
+
+residents_service = ResidentsService(current_app.logger)
+blueprint = Blueprint("residents", __name__, url_prefix="/residents")
+
+
+@blueprint.route("/", methods=["POST"], strict_slashes=False)
+@require_authorization_by_role({"Admin"})
+def add_resident():
+    """
+    Add a resident
+    """
+    resident = request.json
+    try:
+        created_resident = residents_service.add_resident(resident)
+        return jsonify(created_resident), 201
+    except Exception as e:
+        error_message = getattr(e, "message", None)
+        return jsonify({"error": (error_message if error_message else str(e))}), 500
+    
+
+@blueprint.route("/<int:resident_id>", methods=["PUT"], strict_slashes=False)
```

Comment:
tbh I don't really know why after changing the table name, when I change this part to residents_id, I get an error